### DISCUSSION TITLE
Very little visibility of member's codespaces, for an organisation owner

### DISCUSSION THREAD

**Discussion by rob-mcgrail** [community/community#12008] [Codespaces]
Hi,

Are there any plans to give organisation or enterprise admins more visibility on codespaces usage? The information that appears under "billing" is very sparse. 

If you have a team of 10 developers you want to be able to see if someone has created 14 codespaces for the same repo, or be able to remove codespaces if there are 15 old ones using large amounts of storage on an old project etc.

At the moment it seems like all you can see is events in the audit log, and your own personal codespaces at the repository level.

Even just being able to see other member's codespaces, on a per-repository basis, would be enough. But ideally I'd like to be able to see every codespace for my organisation, their current status, and have the ability to remove / stop them.

**Comment by tanmayeekamath** [comment]
Hey @rob-mcgrail 
Currently you can get a breakdown of codespaces usage for your org by downloading the usage report: https://docs.github.com/en/billing/managing-billing-for-github-codespaces/viewing-your-codespaces-usage#viewing-codespaces-usage-for-your-organization

Besides that we will soon be adding support for retention policies wherein as an org admin, you will be able to specify how long should unused codespaces created within your org be retained. More info here: https://github.com/github/roadmap/issues/339
Would love to understand if this would address your need?

**Reply by rob-mcgrail** [reply]
Hi,

Retention rules will definitely be reassuring. The billing view is pretty sparse (no repo or member breakdown, possibly this exists in the CSV).

They're both useful, but no real substitute for being able to see "codespaces in my organisation" under a repos codespaces list.

If I'm being billed for them, being able to see them seems a pretty logical feature.

(Understand that having actual access to them would be bad, and that allowing view / delete for them probably complicates preventing that, but please consider it as a feature for the future).

**Comment by sayhiben** [comment]
Adding on here - my organization would like to create a GitHub Action that sends Slack notifications to users who have had a Codespace active for more than ~10 hours (a la Netflix, "Are you still watching?"). It seems that there's not quite enough information available to the enterprise to accomplish this:
- The detailed billing log mentioned above isn't programmatically accessible; it runs a background task that emails a link to the result, which would then have to be polled for, fetched, parsed, etc. It also only aggregates usage on a daily basis.
- The audit log is promising, but does not include `codespaces:*` events that indicate when a Codespace has been suspended

Ideally, I'd like an endpoint to list all organization-owned Codespaces with instance type, repo, user, created datetime, and current session time.

**Comment by Romilpxtel** [comment]
print("Hello, World")
